File: python/src/main/python/stores/MilkIceCreamStore.py
from icecream.milk.GuavaMilkIceCream import *
from icecream.milk.MangoMilkIceCream import *
from icecream.milk.PineappleMilkIceCream import *
from icecream.milk.StrawberryMilkIceCream import *
from IceCreamStore import *


class MilkIceCreamStore(IceCreamStore):
    def createIceCream(self, flavor):
        if (flavor == "Strawberry"):
            return StrawberryMilkIceCream()
        elif (flavor == "Pineapple"):
            return PineappleMilkIceCream()
        elif (flavor == "Mango"):
            return MangoMilkIceCream()
        elif (flavor == "Guava"):
            return GuavaMilkIceCream()
        return None

File: csharp/factorypattern/main/stores/MilkIceCreamStore.cs
using factorypattern.main.icecreams;
using factorypattern.main.icecreams.milk;

namespace factorypattern.main.stores
{
    internal class MilkIceCreamStore : IceCreamStore
    {
    override
     protected IceCream createIceCream(string flavor)
        {
            if (flavor.Equals("Strawberry"))
            {
                return new StrawberryMilkIceCream();
            }

            if (flavor.Equals("Pineapple"))
            {
                return new PineappleMilkIceCream();
            }
            if (flavor.Equals("Mango"))
            {
                return new MangoMilkIceCream();
            }
            return flavor.Equals("Guava") ? new GuavaMilkIceCream() : null;
        }


    }
}
